### PR TITLE
Suggested changes to Pitot display

### PR DESCRIPTION
**Pull request by KirikauKiwi**
I have made some suggested changes to the Pitot display tile.

- Changes so the ∆ will display -/+ inH20 which is more in keeping with how the other pressure sensors are viewed
- Changed the index.html so that the ∆ decimal display will change with the configuration decimal accuracy
- The velocity will still be displayed as a positive value only.

Tested and working on my setup, the dynamic changes in the ∆ do not effect the tile size, everything stay on a single line.

### CODE REVIEW

**Comment by DeeEmm** [comment]
The delta should only ever be positive. If you have negative delta, you have negative velocity, which should never happen. This was why I reverted the original change so that both velocity and delta are positive

As I mentioned in the other thread, the only way to get a negative deltas is to (incorrectly) zero the probe in the port. The probe should be zeroed in ambient conditions outside of the port.

In short, there is no need for negative delta or velocity. It should not follow other pressure conditions as it is not measuring pressure.

In V3 we will add pressure measurement to the Pitot for single ended measurements, which will be +ve/-ve, but I think that the velocity display should remain as-is.

**Comment by KirikauKiwi** [comment]
While testing the pitot it feels more natural to me to see the negative value. i.e less confusing as you know exactly what the sensor is doing.
- It shows wether the sensor is either sitting above or below the crossing point before calibrating or zeroing it.
- I can see it being helpful for trouble shooting, if people are having problems with their pitot measurements it will allow you to see what way their pitot is connected without having to enquire.
- It shows the true reading of the sensor.
- It allows

I agree with you about not having a negative delta to a point that it is absolutely the correct way of doing it (this is only possible because of the sensor type), but if it was to be done 100% correctly you would have to ensure that the end user is correctly connecting their Pitot probe to the sensor too. e.g. P_total should always be connected to the P1 port of sensor and P_static to the P2 port.

In either the current method we are allowing the Pitot probe to be connected in reverse and masking it in the code, with my method it still allows the reverse connection of the probe but it is visible by having a negative delta, and then masking in code the velocity.

Either way works, as you have said before "many ways to skin a cat" either method is a compromise, the most correct way would be to have P_total connected to the P1 port of the sensor and the absolutes for the values removed from the formulas.

Regarding the single port probe and the +/-, should it really be a +/- when displayed as it will still be a positive pressure being measured, which is useful but if you were to graph it you would need to know the base value that it was zeroed from.

Secondly if someone was working on a cylinder port how are they going to know what their reference pressure was to then be able to compare it to a second measurement after modifying the cylinder port.
I would think it would be preferable not to be able to zero this measurement as you are always going to have a true/comparable reading between measurements and therefore be able to see a gain or loss, where as if the it was zeroed in the port you will only be able to see the deltas in the port for that one measurement.

**Comment by DeeEmm** [comment]
> While testing the pitot it feels more natural to me to see the negative value. i.e less confusing as you know exactly what the sensor is doing.

It should be considered that pitot velocity is an **absolute measurement**. It will **never** be negative. We are specifically talking about the velocity measurement here. Not pressure - which is something altogether different. Sure there is a pressure component to the velocity measurement - the delta differential, but that should not be viewed as a pressure measurement in the same way as pDiff or using the Pitot sensor single ended. If you are expecting to see the value dance around the crossover point it will not. That's not how it works. If that's your expectation then you will need to use the single ended probe in pressure mode.

With the probe outside of the port you will see zero flow, but as soon as you put it into the port, it will give you a positive flow reading. Move it around the port and that reading will be higher or lower, but it will **always** be positive,

With the pitot probe connected it will never give you a negative delta differential value, simply as the probe port ~should~ **will** never be lower than the static port. It is just the way that the probe works.

Think of it like the pitot tube on an aeroplane. The pressure tube always points forwards. If you get a negative value you've got BIG problems as it means that you are flying backwards. It's exactly the same thing here. It will NEVER read negative. There's no reversion on a flow bench.

So just as for ANY sensor in ANY application, Out of range measurements are not shown. Simply as they are 'out of range'.

> * It shows wether the sensor is either sitting above or below the crossing point before calibrating or zeroing it.
> * I can see it being helpful for trouble shooting, if people are having problems with their pitot measurements it will allow you to see what way their pitot is connected without having to enquire.

This does make some sense, but in this case you just switch to the pressure reading to view the -ve value. When we zero the pitot,  we are zeroing the sensor differential. Which is the same no matter what is connected to it or what is displaying in the GUI. The only time that this can be stuffed up is if the sensor / probe / etc is not zeroed in ambient conditions. In which case there's really no helping 🤣 

If there _is_ an issue, then fault finding should not really be done via the GUI but by using the raw data available in the API / serial monitor, which is the original intent. There are plans down the track to make API data available in the GUI via a serial console. It should be possible to diagnose any issue via the API calls.

Remembering also that there is no error detection in place at the moment. If we are seeing a non positive sensor value then we can raise an error and report it to the GUI. i.e. 'Pitot out of range'. Same as any other critical operational error. This is something that is on the roadmap but needs to be added in.

> In either the current method we are allowing the Pitot probe to be connected in reverse and masking it in the code, with my suggestions it still allows the reverse connection of the probe but it is visible by having a negative delta, and then masking in code the velocity.

It is not masking an issue, as there is no issue. The pressure sensor has +ve/-ve range, so we just allow the probe to be connected either way, it's not specifically connected in forward or reverse. It actually makes things easier for the end user. I can remove the fabs() command to force people to have to connect their pitot with the probe por to P1 and the static port to P2, but it is unnecessary, especially to justify adding a -ve reading to the velocity tile. I definitely do not see the current configuration as a compromise. It does exactly what it states on the tin. It displays velocity.

Just remember that the pitot velocity is an absolute measurement and can never be negative and you will see that there is no point / reason for displaying the negative value. Same with the delta differential. The probe port will ALWAYS be higher than the static pressure port.

**Comment by KirikauKiwi** [comment]
Keep it how it is regarding the velocity and pressure.

Please consider the suggested change to allow the decimal display for inH2O to follow the General Decimal accuracy from the configuration page.

**Comment by DeeEmm** [comment]
I don't understand the second commit. It seems to be reversing an earlier commit?

Screenshot 2024-12-01 at 10 53 15 pm

Screenshot 2024-12-01 at 10 53 43 pm

**Comment by KirikauKiwi** [comment]
The second one is related to me doing the PR from the wrong branch at my end!!!!
The one to look at is PR #272 it has Pitot display, the incorrect one was from PR#217 and is labeled as Pitot Display.
I am not sure how they have ended up together in #272

**Comment by DeeEmm** [comment]
I've had some weird stuff going on with my VSCode as well.

Are you running multiple instances with different branches? I'm wondering if this is causing the issues I'm seeing. Is fine most of the time, but every now and then has a tizzy-fit

**Comment by KirikauKiwi** [comment]
That is odd, I didn't create a pull request for this, only sync'd my local branch with my remote branch.
Still learning to drive Github correctly. Will try and work out how not to do that again.

**Comment by DeeEmm** [comment]
All good. It's never quite been the same since Micro$oft took over 😆

**Comment by KirikauKiwi** [comment]
I think it is because I am using the same remote branch for the changes at my end, it looks like I need to make a new branch for each change at my end.

**Comment by DeeEmm** [comment]
Yes this is the preferred way.

If your change relates to an issue, you can create the issue here within the project, and then a dedicated branch for the issue can be created.

Or, if you are just working remote and want to forego creating the issue here, then create a new branch that is based on DEV with the changes.

**Comment by KirikauKiwi** [comment]
I have closed this as I have submitted an update to this pull request here PR #278
